hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieCompactor.java

```diff
@@ -270,6 +270,7 @@ private int doCompact(JavaSparkContext jsc) throws Exception {
         }
       }
       HoodieWriteMetadata<JavaRDD<WriteStatus>> compactionMetadata = client.compact(cfg.compactionInstantTime);
+      client.commitCompaction(cfg.compactionInstantTime, compactionMetadata, Option.empty());
```

Comment:
Maybe we add inline compaction method.

Reply:
yes, we could do that.

Reply:
actually, for external jobs like this one, we don't wanna use auto commit flows. lets keep it this way. 
I have fixed Metadata writer to just commit with `writeClient.compact()` call only w/o needing additional `wirteClient.commitCompaction`. 
for other external jobs like these, lets allow only non auto commit flows.

Reply:
I'm not sure if the `commitXXX` plus `completeXXX` combination as public API makes sense in general, because from intuition, the `commitXXX` already means to complete the commit. Inversely I kind of think the `completeXXX` should be deemed as a internal API to support tests/debugging or some cases that needs flexibility.

I know Spark has issues/feature like lazy triggering of computation of RDDs, but let's make our API semantics clear first.

Reply:
the had to leave the other API mainly for flink.
but I agree w/ you. we should keep it tight. just 1 of commitXXX or completeXXX should suffice as public